src/main/java/cms/gov/madie/measure/services/MeasureService.java

```diff
@@ -36,6 +44,57 @@ public class MeasureService {
   private final MeasureRepository measureRepository;
   private final ElmTranslatorClient elmTranslatorClient;
   private final MeasureUtil measureUtil;
+  private final ActionLogService actionLogService;
+  private final TerminologyValidationService terminologyValidationService;
+
+  public Measure createMeasure(Measure measure, final String username, String accessToken) {
+    log.info("User [{}] is attempting to create a new measure", username);
+    checkDuplicateCqlLibraryName(measure.getCqlLibraryName());
+    validateMeasurementPeriod(
+        measure.getMeasurementPeriodStart(), measure.getMeasurementPeriodEnd());
+    Measure measureCopy = measure;
```

Comment:
This does not create copy. It creates another pointer targeting the same object.

Reply:
a copy was not needed but codacy was complaining about reassigning method parameters. So, just made up something. `Measure measureCopy = measure.toBuilder().build();` would have been better but facing some issues with test cases. The builder creates a singleton collection for errors object which can't be modified and then assertions fails.